services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/map/ProcessedMapRepositoryImpl.java

```diff
@@ -0,0 +1,289 @@
+import com.mongodb.client.model.Projections;
+import com.mongodb.client.model.Sorts;
+
+
+import static com.mongodb.client.model.Filters.eq;
+
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.map.ProcessedMap;
+import us.dot.its.jpo.ode.api.ConflictMonitorApiProperties;
+import us.dot.its.jpo.ode.api.models.IDCount;
+import us.dot.its.jpo.ode.api.models.IntersectionReferenceData;
+import us.dot.its.jpo.conflictmonitor.monitor.models.map.MapBoundingBox;
+import us.dot.its.jpo.conflictmonitor.monitor.models.map.MapIndex;
+import us.dot.its.jpo.geojsonconverter.DateJsonMapper;
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.LineString;
+
+@Component
+public class ProcessedMapRepositoryImpl implements ProcessedMapRepository {
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    ConflictMonitorApiProperties props;
+
+    TypeReference<ProcessedMap<LineString>> processedMapTypeReference = new TypeReference<>(){};
+
+    private String collectionName = "ProcessedMap";
+    private ObjectMapper mapper = DateJsonMapper.getInstance();
+    private Logger logger = LoggerFactory.getLogger(ProcessedMapRepositoryImpl.class);
+
+    public Query getQuery(Integer intersectionID, Long startTime, Long endTime, boolean latest, boolean compact) {
+        Query query = new Query();
+
+        if (intersectionID != null) {
+            query.addCriteria(Criteria.where("properties.intersectionId").is(intersectionID));
+        }
+
+        String startTimeString = Instant.ofEpochMilli(0).toString();
+        String endTimeString = Instant.now().toString();
+
+        if (startTime != null) {
+            startTimeString = Instant.ofEpochMilli(startTime).toString();
+        }
+        if (endTime != null) {
+            endTimeString = Instant.ofEpochMilli(endTime).toString();
+        }
+
+        if (latest) {
+            query.with(Sort.by(Sort.Direction.DESC, "properties.timeStamp"));
+            query.limit(1);
+        }else{
+            query.limit(props.getMaximumResponseSize());
+        }
+
+        if (compact){
+            query.fields().exclude("recordGeneratedAt", "properties.validationMessages");
+        }else{
+            query.fields().exclude("recordGeneratedAt");
+        }
+
+        query.addCriteria(Criteria.where("properties.timeStamp").gte(startTimeString).lte(endTimeString));
+        return query;
+    }
+
+    public long getQueryResultCount(Query query) {
+        return mongoTemplate.count(query, ProcessedMap.class, collectionName);
+    }
+
+    public long getQueryFullCount(Query query){
+        int limit = query.getLimit();
+        query.limit(-1);
+        long count = mongoTemplate.count(query, ProcessedMap.class, collectionName);
+        query.limit(limit);
+        return count;
+    }
+
+    public List<ProcessedMap<LineString>> findProcessedMaps(Query query) {
+        List<Map> documents = mongoTemplate.find(query, Map.class, collectionName);
```

Comment:
question: is there a benefit to querying for generic Map type and converting manually instead of querying for the `ProcessedMap<Linestring>` that we're returning?

Reply:
The _id field needs to be removed before the object can be cast/converted into a ProcessedMap object

Reply:
If the mongoTemplate really isn't smart enough to ignore extra fields, we could add `query.fields().exclude("_id")` so we don't have to iterate over the returned collection.

Reply:
Just pushed an update setting FAIL_ON_UNKNOWN_PROPERTIES to false on the object mapper and removing the manual _id removal

Reply:
I did some testing using ProcessedMap as the output class from MongoTemplate.find, and it is dropping the "id" fields, which identify the lanes within the message... I created a task item to investigate converting this to complete all conversion using mongo and not ObjectMapper

Reply:
Here is that work item: https://dev.azure.com/SOC-OIT/CDOT/_backlogs/backlog/CDOT%20Connected%20Vehicles/Backlog%20items/?workitem=290061